File: binary_search/javascript/binary_search.js
function binary_search(arr,element){
    var left=0;
    var right=arr.length-1;
    while(right>=left){
        var mid=Math.floor(left+(right-left)/2);
        if(arr[mid]==element)
            {return mid;}
        if(arr[mid]>element)
           { right=mid-1;}
        else
            {left=mid+1;}
        
    }
    return false;
}

var arr=[50,100,60,85,25,30,95,158,563,1000,5,9,7];
var element=60;
//binary search works on a sorted array

arr.sort((a,b)=>a-b);
binary_search(arr,element)?console.log("Data Found"):console.log("Data Not Found");

File: binary_search/python/binarysearch.py
def binary(l,k): 
 lb=0
 ub=len(l)-1
 while ub<lb:
  mid=(ub+lb)//2
  if l[mid]<k:
   lb=mid+1
  elif l[mid]>k:
   ub=mid
  else:
    return(mid)
 return -1
  
print("Enter the number of elements")
n=int(input())
l=[0]*n
ans=-1
print("Enter the sorted list of elements")
for i in range(n):
 l[i]=int(input())
print("Enter the element to search")
k=int(input())
r=0
r=binary(l,k)
if r!=-1:
    print("element is present at -")
    print(r)
else:
    print("not present")


